llvm-external-projects/iree-dialects/lib/Transforms/TransformMatchers.cpp

```diff
@@ -499,34 +501,33 @@ transform_ext::StructuredOpMatcher &transform_ext::StructuredOpMatcher::input(
 transform_ext::StructuredOpMatcher &transform_ext::StructuredOpMatcher::input(
     int64_t position, SameOperandAsProducer producerPosition) {
   predicates.push_back([=](linalg::LinalgOp linalgOp) -> bool {
-    LLVM_DEBUG(DBGS() << "input operand " << position
-                      << " is the same value as operand "
-                      << producerPosition.value.second << " of the input "
-                      << producerPosition.value.first << " producer");
-    int64_t updatedPosition =
-        position >= 0 ? position : linalgOp.getNumDpsInputs() + position;
-    if (0 > updatedPosition || updatedPosition >= linalgOp.getNumDpsInputs())
-      return false;
-    int64_t updatedProducerPosition =
-        producerPosition.value.first >= 0
-            ? producerPosition.value.first
-            : linalgOp.getNumDpsInputs() + producerPosition.value.first;
-    if (0 > updatedProducerPosition ||
-        updatedProducerPosition >= linalgOp.getNumDpsInputs())
-      return false;
-    Operation *producer = linalgOp.getDpsInputOperand(updatedProducerPosition)
-                              ->get()
-                              .getDefiningOp();
-    if (!producer)
-      return false;
+    // LLVM_DEBUG(DBGS() << "input operand " << position
+    //                   << " is the same value as operand "
+    //                   << producerPosition.value << " of the input "
+    //                   << producerPosition.value << " producer");
+    linalg::LinalgOp producerOp = linalgOp;
+    for (size_t i = 0, e = producerPosition.value.size() - 1; i < e; i++) {
```

Comment:
Nit: something like `int64_t pos : llvm::drop_front(producerPosition.value)` would be more idiomatic here.